src/NHibernate/Event/Default/DefaultMergeEventListener.cs

```diff
@@ -230,9 +230,6 @@ private object MergeTransientEntity(object entity, string entityName, object req
 			else
 			{
 				copy = source.Instantiate(persister, id);
-				// #1436: we use the persister to instanciate a new, transient entity. In case it has lazy properties,
-				// we need to to tell its proxy that the properties are all fetched, since it is a transient entity.
-				persister.AfterInitialize(copy, false, source);
```

Comment:
Fix no more needed with the alternate fix. Not having the proxy initialized is then equivalent to having it considering all properties are already fetched.

Ascertained by adding more asserts in the transient merge test.